static/js/settings_org.js

```diff
@@ -277,18 +277,28 @@ exports.populate_realm_domains = function (realm_domains) {
     });
 };
 
+function sort_object_by_key(obj) {
+    var keys = _.keys(obj).sort();
+    var new_obj = {};
+    _.each(keys, function (key) {
+        new_obj[key] = obj[key];
+    });
+    return new_obj;
+}
+
 exports.populate_auth_methods = function (auth_methods) {
     if (!meta.loaded) {
         return;
     }
 
     var auth_methods_table = $("#admin_auth_methods_table").expectOne();
     auth_methods_table.find('tr.method_row').remove();
-    _.each(_.keys(auth_methods).sort(), function (key) {
+    auth_methods = sort_object_by_key(auth_methods);
+    _.each(auth_methods, function (value, auth_method) {
         auth_methods_table.append(templates.render('admin_auth_methods_list', {
             method: {
-                method: key,
-                enabled: auth_methods[key],
+                method: auth_method,
+                enabled: value,
```

Comment:
Can you explain this commit?  I feel like the new code is less readable than the old code, so I'm curious why you made this change.

Reply:
I extracted `sort_obj_by_key` function because we used it afterward also to sort objects by key, therefore extracted it and made it what it is now.

Reply:
I've explained my above comment more clearly here https://github.com/zulip/zulip/pull/12070/files#r281007229, let me know if this makes sense now!